cpp/src/filling/fill.cu

```diff
@@ -60,11 +61,38 @@ struct in_place_fill_range_dispatch {
   cudf::mutable_column_view& destination;
 
   template <typename T>
-  std::enable_if_t<cudf::is_fixed_width<T>(), void> operator()(cudf::size_type begin,
+  std::enable_if_t<cudf::is_fixed_width<T>() && not cudf::is_fixed_point<T>(), void> operator()(
+    cudf::size_type begin, cudf::size_type end, cudaStream_t stream = 0)
+  {
+    in_place_fill<T>(destination, begin, end, value, stream);
+  }
+
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_point<T>(), void> operator()(cudf::size_type begin,
                                                                cudf::size_type end,
                                                                cudaStream_t stream = 0)
   {
-    in_place_fill<T>(destination, begin, end, value, stream);
+    if (value.type().id() == cudf::type_id::DECIMAL32) {
```

Comment:
This `if/else` should not be necessary. There should only be a single code path that can query all of the necessary information from `T`.

Reply:
@jrhemstad It's done. But I am not quite sure whether my change is right. But  test cases are happy to it at least.